base_location/models/company.py

```diff
@@ -6,19 +6,46 @@
 
 
 class ResCompany(models.Model):
-
     _inherit = 'res.company'
 
-    @api.onchange('better_zip_id')
-    def on_change_city(self):
-        if self.better_zip_id:
-            self.zip = self.better_zip_id.name
-            self.city = self.better_zip_id.city
-            self.state_id = self.better_zip_id.state_id
-            self.country_id = self.better_zip_id.country_id
+    city_id = fields.Many2one(
+        'res.city',
+        compute='_compute_address',
+        inverse='_inverse_city_id',
+        string="City"
+    )
 
-    better_zip_id = fields.Many2one(
+    zip_id = fields.Many2one(
         'res.better.zip',
-        string='Location',
+        string='ZIP Location',
+        compute='_compute_address',
```

Comment:
This should be a simply related field to `partner_id.zip_id`. Don't complicate things...

Reply:
We are following the same configuration used in [odoo company](https://github.com/odoo/odoo/blob/11.0/odoo/addons/base/res/res_company.py#L59)

Reply:
If I'm not wrong, `res.company` extends `res.partner`, so `zip_id` should be available out of the box, isn't it?